### DISCUSSION TITLE
Problems installing IRAF in Intel Macbook (macOS Monterey)

### DISCUSSION THREAD

**Discussion by miriamram96** [iraf-community/iraf#232] [Q&A] [answered]
Hi, I have been struggling with my IRAF installation. I have been using the terminal in my Intel MacBook Pro. When I try to run IRAF I get the following errors: Bad CPU in executable and Undefined error:0.

   ```
    (base) Miriams-MacBook-Pro:~ user$ source activate iraf27    
    (iraf27) Miriams-MacBook-Pro:~ user$ cd iraf
    (iraf27) Miriams-MacBook-Pro:iraf user$ mkiraf
    Initialize uparm? (y|n): y
    Terminal types: xgterm,xterm,gterm,vt640,vt100,etc.
    Enter terminal type [default xterm-256color]: xterm
    A new LOGIN.CL file has been created in the current directory.
    You may wish to review and edit this file to change the defaults.
    You may also add additional customizations to loginuser.cl
    (iraf27) Miriams-MacBook-Pro:iraf user$ cl
    /Users/user/anaconda3/envs/iraf27/bin/cl: line 40:         
    /Users/user/anaconda3/envs/iraf27/iraf//bin.macosx/ecl.e: Bad CPU type in executable
    /Users/user/anaconda3/envs/iraf27/bin/cl: line 40: /Users/user/anaconda3/envs/iraf27/iraf//bin.macosx/ecl.e: Undefined error: 0
```

When I open /Users/user/anaconda3/envs/iraf27/iraf//bin.macosx/ecl.e: in my Jupiter Notebook, it shows the following:

     Error! /Users/user/anaconda3/envs/iraf27/iraf//bin.macosx/ecl.e is not UTF-8 encoded.
     Saving disabled.
     See Console for more details. 

Anybody knows how to deal with this issue? Thank you!

**Comment by olebole** [comment, marked as answer]
The reason is that the astroconda IRAF package is 32-bit only and 32-bit executables are no longer supported by macOS. On your computer, you need to compile IRAF yourself, as described in https://iraf-community.github.io/install.html.
